### ISSUE TITLE
Flatpak build

### ISSUE THREAD

**Issue by LeandroStanger** [Vita3K/Vita3K#1418]
Please, consider supporting [Flatpak](http://docs.flatpak.org/en/latest/introduction.html#reasons-to-use-flatpak) and publishing Vita3K on [Flathub](https://github.com/flathub/flathub/wiki/App-Submission).

**Comment by Doomsdayrs** [comment]
Second, An ubuntu-tar is irrelevant and a flatpak bundle can be used on more systems

**Comment by Doomsdayrs** [comment]
Here is the way https://docs.flatpak.org/en/latest/single-file-bundles.html

**Comment by qurious-pixel** [comment]
It is more than just bundling the app. 
flatpak has some read-only portions that Vita3K currently does not support.
It will fail to launch.

**Comment by Doomsdayrs** [comment]
> It is more than just bundling the app. flatpak has some read-only portions that Vita3K currently does not support. It will fail to launch.

What read-only portions?

**Comment by qurious-pixel** [comment]
https://docs.flatpak.org/en/latest/sandbox-permissions.html
Sandbox permissions prevent the creation of `shaderlog, config.yml, and vita3k.log`
The root folder of `data, shaders, shaders-builtin and Vita3k` are in a read-only sandbox when packaged as a flatpak.

**Comment by Doomsdayrs** [comment]
You can drop a payload into the user data directory.

This is how the flatpak for WarThunder works

**Comment by EXtremeExploit** [comment]
> https://docs.flatpak.org/en/latest/sandbox-permissions.html
> Sandbox permissions prevent the creation of `shaderlog, config.yml, and vita3k.log`
> The root folder of `data, shaders, shaders-builtin and Vita3k` are in a read-only sandbox when packaged as a flatpak.

This could be implemented to go into ./config or ./local with the right permissions and PR, still its not very good to have the emularor "unbox" just for these files

**Comment by qurious-pixel** [comment]
This can be accomplished by placing the appropriate files/folders in `$XDG_DATA_HOME` and `$XDG_CONFIG_HOME`, but this will also affect Windows and MacOS.
I prefer this for Linux (it follows the XDG standard), but would require Windows users to look in `%LOCALAPPDATA%`

**Comment by Doomsdayrs** [comment]
If there is a requirement for having it be portable for windows / MacOS, then there can just be a different variant with it set to portable mode.

**Comment by P-C-R** [comment]
+1 for a flatpak build. For all the SteamDeck users out there

**Comment by j8r** [comment]
Hello, I'm working on packaging Vita3K for Flatpak. Good new, I got the package built! However, work remains.

First, a few things regarding the compilation:
- libstdc++ is missing on the Flatpak SDK, which makes the final linking fail, and it is not mandatory if we remove it from https://github.com/Vita3K/Vita3K/blob/master/vita3k/CMakeLists.txt#L139
- the bundled boost compiles the `atomic,chrono,exception,filesystem,headers,locale,log,program_options,regex,system,test,thread,timer` libraries, but only  `filesystem,system` are necessary for the compilation
- loads of things could be avoided to be compiled, like googletest and .a static archives, but this is not a big deal, they can be stripped away afterwards
- `-DCMAKE_INTERPROCEDURAL_OPTIMIZATION` could be considered to potentially improve performance. Some other emulators like Citra, DuckStation and PCSX2. It would be interesting to see what difference it will make, if any.

As already said above by @qurious-pixel Vita3K will have to be adapted a bit to run properly inside Flatpak, but at least it will be a starting point.
When running Vita3K, I got this error:
```sh
flatpak run org.vita3k.Vita3K
File log initialization failed: Failed opening file /app/bin/vita3k.log for writing: Read-only file system
```
A solution is either outputting to the stdout directly, or writing to `$XDG_DATA_DIR`.

Here is the [Flatpak](https://github.com/j8r/flathub/tree/org.vita3k.Vita3K), with the build instructions in the README for those wanting to help on making Vita3K running properly inside the sandbox.

**Comment by j8r** [comment]
Hello, I'm working on packaging Vita3K for Flatpak. Good news, I got the package built! However, work remains.

First, a few things regarding the compilation:
- libstdc++ is missing on the Flatpak SDK, which makes the final linking fail, and it is not mandatory if we remove it from https://github.com/Vita3K/Vita3K/blob/master/vita3k/CMakeLists.txt#L139
- the bundled boost compiles the `atomic,chrono,exception,filesystem,headers,locale,log,program_options,regex,system,test,thread,timer` libraries, but only  `filesystem,system` are necessary for the compilation
- loads of things could be avoided to be compiled, like googletest and .a static archives, but this is not a big deal, they can be stripped away afterwards
- `-DCMAKE_INTERPROCEDURAL_OPTIMIZATION` could be considered to potentially improve performance. Some other emulators like Citra, DuckStation and PCSX2. It would be interesting to see what difference it will make, if any.

As already said above by @qurious-pixel Vita3K will have to be adapted a bit to run properly inside Flatpak, but at least it will be a starting point.
When running Vita3K, I got this error:
```sh
flatpak run org.vita3k.Vita3K
File log initialization failed: Failed opening file /app/bin/vita3k.log for writing: Read-only file system
```
A solution is either outputting to the stdout directly, or writing to `$XDG_DATA_DIR`.

Here is the [Flatpak](https://github.com/j8r/flathub/tree/org.vita3k.Vita3K), with the build instructions in the README for those wanting to help on making Vita3K running properly inside the sandbox.

**Comment by j8r** [comment]
I haven't opened a PR because the application was not ready. It has to work before submitting to the store.

**Comment by WingofaGriffin** [comment]
Apologies, I was unclear. I was referring to a PR for the missing dependencies to be added to the SDK.

**Comment by WingofaGriffin** [comment]
Additionally @qurious-pixel I'm curious about the status of your AppImage fork. Are the steps outlined by j8r similar, or does AppImage have different requirements that should be outlined in a separate issue?

**Comment by powerpuffboysz** [comment]
> I haven't opened a PR because the application was not ready. It has to work before submitting to the store.

Any news on the vita flatpak?

**Comment by j8r** [comment]
If anyone has the time to work on making Vitak3K work with the XDG spec, it will unblock the packaging.

**Comment by j8r** [comment]
With the recent XDG changes, I have updated my Flatpak package but got this compiling error with GCC 13:
```
FAILED: vita3k/renderer/CMakeFiles/renderer.dir/src/vulkan/context.cpp.o 
/usr/bin/ccache /run/ccache/bin/c++ -DBOOST_ATOMIC_NO_LIB -DBOOST_FILESYSTEM_NO_LIB -DBOOST_SYSTEM_NO_LIB -DONLY_MSPACES=1 -DSPDLOG_COMPILED_LIB -DSPDLOG_FMT_EXTERNAL -DSPDLOG_NO_THREAD_ID -DTRACY_ENABLE -DTRACY_NO_SYSTEM_TRACING -DTRACY_ON_DEMAND -DUSE_LOCK=0 -DYAML_CPP_STATIC_DEFINE -I/run/build/vita3k/_flatpak_build/vita3k/config -I/run/build/vita3k/vita3k/renderer/include -I/run/build/vita3k/vita3k/crypto/include -I/run/build/vita3k/vita3k/display/include -I/run/build/vita3k/vita3k/emuenv/include -I/run/build/vita3k/vita3k/mem/include -I/run/build/vita3k/vita3k/util/include -I/run/build/vita3k/vita3k/config/include -I/run/build/vita3k/vita3k/np/include -I/run/build/vita3k/external/yaml-cpp/include -I/run/build/vita3k/external/fmt/include -I/run/build/vita3k/external/spdlog/include -I/run/build/vita3k/vita3k/http/include -I/run/build/vita3k/vita3k/kernel/include -I/run/build/vita3k/vita3k/rtc/include -I/run/build/vita3k/vita3k/cpu/include -I/run/build/vita3k/vita3k/nids/include -I/run/build/vita3k/external/dlmalloc -I/run/build/vita3k/external/stb -I/run/build/vita3k/vita3k/shader/include -I/run/build/vita3k/vita3k/features/include -I/run/build/vita3k/vita3k/gxm/include -I/run/build/vita3k/vita3k/glutil/include -I/run/build/vita3k/external/glad/include -I/run/build/vita3k/vita3k/threads/include -I/run/build/vita3k/vita3k/vkutil/include -I/run/build/vita3k/external/Vulkan-Headers/include -I/run/build/vita3k/external/VulkanMemoryAllocator-Hpp/include -I/run/build/vita3k/external/VulkanMemoryAllocator-Hpp/VulkanMemoryAllocator/include -I/run/build/vita3k/external/ddspp -I/usr/include/SDL2 -I/run/build/vita3k/external/ffmpeg/include -I/run/build/vita3k/external/xxHash/cmake_unofficial/.. -isystem /run/build/vita3k/external/boost -isystem /run/build/vita3k/external/tracy/public -O2 -pipe -g -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -O2 -g -DNDEBUG -std=c++20 -Wformat -Werror=format-security -MD -MT vita3k/renderer/CMakeFiles/renderer.dir/src/vulkan/context.cpp.o -MF vita3k/renderer/CMakeFiles/renderer.dir/src/vulkan/context.cpp.o.d -o vita3k/renderer/CMakeFiles/renderer.dir/src/vulkan/context.cpp.o -c /run/build/vita3k/vita3k/renderer/src/vulkan/context.cpp
/run/build/vita3k/vita3k/renderer/src/vulkan/context.cpp: In member function ‘void renderer::vulkan::VKContext::start_render_pass(bool)’:
/run/build/vita3k/vita3k/renderer/src/vulkan/context.cpp:298:5: error: ambiguous overload for ‘operator=’ (operand types are ‘vk::RenderPassBeginInfo’ and ‘<brace-enclosed initializer list>’)
  298 |     };
```

By the way, is it ok to compile Vita3K with GCC or would it be better - i.e. faster and/or more supported - to use LLVM?

**Comment by theofficialgman** [comment]
> By the way, is it ok to compile Vita3K with GCC or would it be better - i.e. faster and/or more supported - to use LLVM?

@j8r just another user here but clang 15 is used in the official CI https://github.com/Vita3K/Vita3K/blob/master/.github/workflows/c-cpp.yml

**Comment by Macdu** [comment]
> With the recent XDG changes, I have updated my Flatpak package but got this compiling error with GCC 13:
> 
> ```
> FAILED: vita3k/renderer/CMakeFiles/renderer.dir/src/vulkan/context.cpp.o 
> /usr/bin/ccache /run/ccache/bin/c++ -DBOOST_ATOMIC_NO_LIB -DBOOST_FILESYSTEM_NO_LIB -DBOOST_SYSTEM_NO_LIB -DONLY_MSPACES=1 -DSPDLOG_COMPILED_LIB -DSPDLOG_FMT_EXTERNAL -DSPDLOG_NO_THREAD_ID -DTRACY_ENABLE -DTRACY_NO_SYSTEM_TRACING -DTRACY_ON_DEMAND -DUSE_LOCK=0 -DYAML_CPP_STATIC_DEFINE -I/run/build/vita3k/_flatpak_build/vita3k/config -I/run/build/vita3k/vita3k/renderer/include -I/run/build/vita3k/vita3k/crypto/include -I/run/build/vita3k/vita3k/display/include -I/run/build/vita3k/vita3k/emuenv/include -I/run/build/vita3k/vita3k/mem/include -I/run/build/vita3k/vita3k/util/include -I/run/build/vita3k/vita3k/config/include -I/run/build/vita3k/vita3k/np/include -I/run/build/vita3k/external/yaml-cpp/include -I/run/build/vita3k/external/fmt/include -I/run/build/vita3k/external/spdlog/include -I/run/build/vita3k/vita3k/http/include -I/run/build/vita3k/vita3k/kernel/include -I/run/build/vita3k/vita3k/rtc/include -I/run/build/vita3k/vita3k/cpu/include -I/run/build/vita3k/vita3k/nids/include -I/run/build/vita3k/external/dlmalloc -I/run/build/vita3k/external/stb -I/run/build/vita3k/vita3k/shader/include -I/run/build/vita3k/vita3k/features/include -I/run/build/vita3k/vita3k/gxm/include -I/run/build/vita3k/vita3k/glutil/include -I/run/build/vita3k/external/glad/include -I/run/build/vita3k/vita3k/threads/include -I/run/build/vita3k/vita3k/vkutil/include -I/run/build/vita3k/external/Vulkan-Headers/include -I/run/build/vita3k/external/VulkanMemoryAllocator-Hpp/include -I/run/build/vita3k/external/VulkanMemoryAllocator-Hpp/VulkanMemoryAllocator/include -I/run/build/vita3k/external/ddspp -I/usr/include/SDL2 -I/run/build/vita3k/external/ffmpeg/include -I/run/build/vita3k/external/xxHash/cmake_unofficial/.. -isystem /run/build/vita3k/external/boost -isystem /run/build/vita3k/external/tracy/public -O2 -pipe -g -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -O2 -g -DNDEBUG -std=c++20 -Wformat -Werror=format-security -MD -MT vita3k/renderer/CMakeFiles/renderer.dir/src/vulkan/context.cpp.o -MF vita3k/renderer/CMakeFiles/renderer.dir/src/vulkan/context.cpp.o.d -o vita3k/renderer/CMakeFiles/renderer.dir/src/vulkan/context.cpp.o -c /run/build/vita3k/vita3k/renderer/src/vulkan/context.cpp
> /run/build/vita3k/vita3k/renderer/src/vulkan/context.cpp: In member function ‘void renderer::vulkan::VKContext::start_render_pass(bool)’:
> /run/build/vita3k/vita3k/renderer/src/vulkan/context.cpp:298:5: error: ambiguous overload for ‘operator=’ (operand types are ‘vk::RenderPassBeginInfo’ and ‘<brace-enclosed initializer list>’)
>   298 |     };
> ```
> 
> By the way, is it ok to compile Vita3K with GCC or would it be better - i.e. faster and/or more supported - to use LLVM?

There is no ci running gcc for Vita3K right now, which can result in some of these errors. Pull request to fix Vita3K on other compilers are accepted without any issue tho. The difference in speed between gcc and clang shouldn't make too much of a difference, but indeed you for sure wouldn't get these errors using clang.

**Comment by j8r** [comment]
Thanks, I'm able to compile Vitak3K with LLVM17 :slightly_smiling_face: 
When I run Vita3K, I got: "Could not create OpenGL context! Does your GPU at least support OpenGL 4.4", then "Emulated environment initialization failed".
I've a HD 5500, and flatpak permissions should be properly set. It might also be a filesystem permission issue by looking at the last lines.
```
[03:45:12.132] |I| [init_config]: backend-renderer: OpenGL
[03:45:12.132] |I| [init_config]: log-level: 0
[03:45:12.195] |I| [main]: Vita3K v0.1.9 1-4ac8a41
[03:45:12.195] |I| [main]: Number of logical CPU cores: 4
[03:45:12.195] |I| [main]: Available ram memory: 15891 MiB
[03:45:12.197] |I| [init]: Base path: /app/bin/
[03:45:12.197] |I| [init]: Static assets path: /home/jrei/.local/share/Vita3K/
[03:45:12.197] |I| [init]: Shared path: /home/jrei/.local/share/Vita3K/
[03:45:12.197] |I| [init]: Log path: /home/jrei/.var/app/org.vita3k.Vita3K/cache/Vita3K/
[03:45:12.197] |I| [init]: User config path: /home/jrei/.var/app/org.vita3k.Vita3K/config/Vita3K/
[03:45:12.197] |I| [init]: User pref path: /home/jrei/.var/app/org.vita3k.Vita3K/data/Vita3K/Vita3K/
[03:45:12.197] |I| [init]: User cache path: /home/jrei/.var/app/org.vita3k.Vita3K/cache/Vita3K/
[03:45:12.273] |I| [create]: GPU = Mesa Intel(R) HD Graphics 5500 (BDW GT2)
[03:45:12.273] |I| [create]: GL_VERSION = 4.6 (Core Profile) Mesa 23.3.0 (git-1fbdd37d4c)
[03:45:12.273] |I| [create]: GL_SHADING_LANGUAGE_VERSION = 4.60
[03:45:12.273] |I| [create]: Your GPU only supports texture barrier, performance may not be good on programmable blending games.
[03:45:12.273] |W| [create]: Consider upgrading to a GPU that has shader interlock.
[03:45:12.273] |E| [load_shaders]: Couldn't open shader: /home/j8r/.local/share/Vita3K/shaders-builtin/opengl/render_main.vert
[03:45:12.273] |E| [load_shaders]: Couldn't open shader: /home/j8r/.local/share/Vita3K/shaders-builtin/opengl/render_main.vert
[03:45:12.273] |C| [init]: Couldn't compile essential shaders for rendering. Exiting
[03:45:12.273] |E| [init]: Failed to initialize screen renderer
```

**Comment by Macdu** [comment]
Do you indeed have a file render_main.vert in /home/j8r/.local/share/Vita3K/shaders-builtin/opengl/ ?

**Comment by j8r** [comment]
Indeed it works by copying this files! However read-only files should be in the Flatpak directly (`/usr/share`, or any other path), for ease-of-use and sandboxing, and those needed to be written created at `XDG_DATA_HOME`.

With this changes i think we are close to something publishable to the Flathub store :slightly_smiling_face:

**Comment by j8r** [comment]
If anyone has an idea how to fix this path issue by using the XDG standard or using a system path, please go ahead!

**Comment by j8r** [comment]
So, I was able to package Vita3K, with some help from @https://github.com/XargonWan 🎉 

<img width="996" height="589" alt="Image" src="https://github.com/user-attachments/assets/5b332892-5bab-4359-9d89-dd36dc6aeab4" />

I'll submit a patch to allow using the shared FFMPEG available by default on the Freedesktop SDK.
On Vita3K side, some directories are still not pointing to their XDG ones, which prevent to have a self-contained Flatpak:
```sh
[01:58:08.840] |E| [load_shaders]: Couldn't open shader: "/home/XXX/.local/share/Vita3K/shaders-builtin/opengl/render_main.vert"
[01:58:08.840] |E| [load_shaders]: Couldn't open shader: "/home/XXX/.local/share/Vita3K/shaders-builtin/opengl/render_main.vert"
...
[02:00:01.088] |E| [init_lang]: Lang file xml not found: "/home/XXX/.local/share/Vita3K/lang/system/en.xml"
[02:00:01.089] |W| [init_font]: Could not find default Vita3K font at "/home/XXX/.local/share/Vita3K/data/fonts", using default ImGui font.
[02:00:01.089] |W| [init_font]: Could not find default Vita3K font at "/home/XXX/.local/share/Vita3K/data/fonts", using default ImGui font.
[02:00:01.089] |W| [init_font]: Could not find default Vita3K font at "/home/XXX/.local/share/Vita3K/data/fonts", using default ImGui font.
```

The code responsible to that looks like to be this one:
https://github.com/Vita3K/Vita3K/blob/703828d56d80f9664e86b512ef8ee0bcaacb5047/vita3k/app/src/app_init.cpp#L246

I moved the `data`, `shaders-builtin` and `lang` directories where the `Vita3K` binary belongs (necessary to have [this condition](https://github.com/Vita3K/Vita3K/blob/703828d56d80f9664e86b512ef8ee0bcaacb5047/vita3k/app/src/app_init.cpp#L243) to pass) , which fixes the issue.

After that, there was this error:
`[17:27:38.928] |E| [init_lang]: Lang file xml not found: "/app/bin/lang/system/en.xml"`
So there is an inconsistency here, this shall use `root_paths.get_base_path() / "lang"` if it exists, and it does in the Flatpak.

I don't find putting this data folders into `/bin` where the Vita3K binary is ideal, that's not really standard. 
A solution could be to to use the first element of `XDG_DATA_DIRS`, if defined.
It could also be set at compile time I suppose? This could allow to share the same logic with the AppImage build.
A simpler solution could be to check if `/usr/share/Vita3K` exists, and if so use that for `data`, `shaders-builtin` and `lang`?

**Comment by EXtremeExploit** [comment]
@j8r Looked at your yml for the flatpak, it seems fine
You are free from publishing it to flathub, even if its not verified it would be awesome to get this packaged into a flatpak
You can also if you want PR the changes to change which ffmpeg build is used
Worth noting that now v3k supports paths inside /usr/share, /usr/local/share, same path as binary, and whatever APPDIR says it is (appimage)
The metainfo can also be PRd
Also keep in mind the PR #3564 which does other linux stuff for flatpak and metainfo

**Comment by fpiesche** [comment]
I have a vita3k flatpak package [on my personal repository](https://github.com/fpiesche/flatpak-builds/blob/main/apps/org.vita3k.vita3k/org.vita3k.vita3k.yml) that's been working fine for my own use for a good while now; feel free to yoink anything you might find useful from that manifest.

Most of the changes to vita3k itself I've made are over on PR #3564 and in [a single patch to use the system-provided version of ffmpeg](https://github.com/fpiesche/flatpak-builds/blob/main/apps/org.vita3k.vita3k/org.vita3k.vita3k.use-system-ffmpeg.patch) if you want to repurpose that @j8r :)

I've got [a decent amount of experience](https://github.com/flathub/flathub/pulls?q=is%3Apr+author%3Afpiesche) with submitting stuff to Flathub by now as well, so would be happy to help out on that front also :)

**Comment by fpiesche** [comment]
Also just as a side note: getting a Flathub distribution "verified" would be as simple as adding a file with a magic authentication token to `vita3k.org/.well-known/org.flathub.VerifiedApps.txt` :)

**Comment by EXtremeExploit** [comment]
I think for verifying i would put it aside until we are 100% sure it can be maintained by someone trustable or is owned by the org, i dont think it would be a problem if its left unverified for some time

**Comment by EXtremeExploit** [comment]
Also considering vita3k doesnt have tagged release builds and builds happen pretty much all the time, how does updating work in it? Will flathub check for updates on the git and build automatically or how would it work?

**Comment by j8r** [comment]
@fpiesche Thanks for your work, unfortunately Vita3K changed since then and no longer compiles after having switched to Qt.
Vita3K only support Qt 6.11, but for now only 6.10 is available in `org.kde.Sdk`.
I tried to compile with 6.10 but got an error.
I also a FFMPEG patch, which I'll be able to submit to upstream.

<details>
<summary>Compilation error</summary>
-- Build files have been written to: /run/build/vita3k/_flatpak_build
[0/2] Re-checking globbed directories...
[391/1239] Building CXX object vita3k/util/CMakeFiles/util.dir/src/string_utils.cpp.o
/run/build/vita3k/vita3k/util/src/string_utils.cpp: In function ‘std::wstring string_utils::utf_to_wide(const std::string&)’:
/run/build/vita3k/vita3k/util/src/string_utils.cpp:31:10: warning: ‘template<class _Codecvt, class _Elem, class _Wide_alloc, class _Byte_alloc> class std::__cxx11::wstring_convert’ is deprecated [-Wdeprecated-declarations]
   31 |     std::wstring_convert<std::codecvt_utf8<wchar_t>> myconv;
      |          ^~~~~~~~~~~~~~~
In file included from /usr/include/c++/15.2.0/locale:47,
                 from /usr/include/c++/15.2.0/format:49,
                 from /usr/include/c++/15.2.0/ostream:44,
                 from /usr/include/c++/15.2.0/istream:43,
                 from /usr/include/c++/15.2.0/sstream:42,
                 from /usr/include/c++/15.2.0/chrono:51,
                 from /run/build/vita3k/external/spdlog/include/spdlog/common.h:10,
                 from /run/build/vita3k/external/spdlog/include/spdlog/spdlog.h:12,
                 from /run/build/vita3k/vita3k/util/include/util/log.h:25,
                 from /run/build/vita3k/vita3k/util/src/string_utils.cpp:20:
/usr/include/c++/15.2.0/bits/locale_conv.h:262:33: note: declared here
  262 |     class _GLIBCXX17_DEPRECATED wstring_convert
      |                                 ^~~~~~~~~~~~~~~
/run/build/vita3k/vita3k/util/src/string_utils.cpp: In function ‘std::string string_utils::wide_to_utf(const std::wstring&)’:
/run/build/vita3k/vita3k/util/src/string_utils.cpp:36:10: warning: ‘template<class _Codecvt, class _Elem, class _Wide_alloc, class _Byte_alloc> class std::__cxx11::wstring_convert’ is deprecated [-Wdeprecated-declarations]
   36 |     std::wstring_convert<std::codecvt_utf8<wchar_t>> myconv;
      |          ^~~~~~~~~~~~~~~
/usr/include/c++/15.2.0/bits/locale_conv.h:262:33: note: declared here
  262 |     class _GLIBCXX17_DEPRECATED wstring_convert
      |                                 ^~~~~~~~~~~~~~~
/run/build/vita3k/vita3k/util/src/string_utils.cpp: In function ‘std::string string_utils::utf16_to_utf8(const std::u16string&)’:
/run/build/vita3k/vita3k/util/src/string_utils.cpp:114:10: warning: ‘template<class _Codecvt, class _Elem, class _Wide_alloc, class _Byte_alloc> class std::__cxx11::wstring_convert’ is deprecated [-Wdeprecated-declarations]
  114 |     std::wstring_convert<std::codecvt_utf8_utf16<char16_t>, char16_t> myconv;
      |          ^~~~~~~~~~~~~~~
/usr/include/c++/15.2.0/bits/locale_conv.h:262:33: note: declared here
  262 |     class _GLIBCXX17_DEPRECATED wstring_convert
      |                                 ^~~~~~~~~~~~~~~
/run/build/vita3k/vita3k/util/src/string_utils.cpp: In function ‘std::u16string string_utils::utf8_to_utf16(const std::string&)’:
/run/build/vita3k/vita3k/util/src/string_utils.cpp:119:10: warning: ‘template<class _Codecvt, class _Elem, class _Wide_alloc, class _Byte_alloc> class std::__cxx11::wstring_convert’ is deprecated [-Wdeprecated-declarations]
  119 |     std::wstring_convert<std::codecvt_utf8_utf16<char16_t>, char16_t> myconv;
      |          ^~~~~~~~~~~~~~~
/usr/include/c++/15.2.0/bits/locale_conv.h:262:33: note: declared here
  262 |     class _GLIBCXX17_DEPRECATED wstring_convert
      |                                 ^~~~~~~~~~~~~~~
[466/1239] Building C object external/CMakeFiles/miniz.dir/miniz/miniz.c.o
/run/build/vita3k/external/miniz/miniz.c:3185:9: note: ‘#pragma message: Using fopen, ftello, fseeko, stat() etc. path for file I/O - this path may not support large files.’
 3185 | #pragma message("Using fopen, ftello, fseeko, stat() etc. path for file I/O - this path may not support large files.")
      |         ^~~~~~~
[707/1239] Building CXX object vita3k/codec/CMakeFiles/codec.dir/src/mp3.cpp.o
/run/build/vita3k/vita3k/codec/src/mp3.cpp: In destructor ‘virtual Mp3DecoderState::~Mp3DecoderState()’:
/run/build/vita3k/vita3k/codec/src/mp3.cpp:205:18: warning: ‘int avcodec_close(AVCodecContext*)’ is deprecated [-Wdeprecated-declarations]
  205 |     avcodec_close(context);
      |     ~~~~~~~~~~~~~^~~~~~~~~
In file included from /run/build/vita3k/vita3k/codec/src/mp3.cpp:21:
/run/build/vita3k/external/ffmpeg/include/libavcodec/avcodec.h:2398:5: note: declared here
 2398 | int avcodec_close(AVCodecContext *avctx);
      |     ^~~~~~~~~~~~~
[844/1239] Building CXX object vita3k/packages/CMakeFiles/packages.dir/src/pkg.cpp.o
/run/build/vita3k/vita3k/packages/src/pkg.cpp: In function ‘bool install_pkg(const boost::filesystem::path&, EmuEnvState&, std::string&, const std::function<void(float)>&)’:
/run/build/vita3k/vita3k/packages/src/pkg.cpp:92:10: warning: ignoring return value of ‘size_t fread(void*, size_t, size_t, FILE*)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
   92 |     fread(reinterpret_cast<void *>(&pkg_header), sizeof(PkgHeader), 1, infile);
      |     ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/run/build/vita3k/vita3k/packages/src/pkg.cpp:94:10: warning: ignoring return value of ‘size_t fread(void*, size_t, size_t, FILE*)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
   94 |     fread(reinterpret_cast<char *>(&ext_header), sizeof(PkgExtHeader), 1, infile);
      |     ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/run/build/vita3k/vita3k/packages/src/pkg.cpp:122:14: warning: ignoring return value of ‘size_t fread(void*, size_t, size_t, FILE*)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  122 |         fread(block, sizeof(block), 1, infile);
      |         ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/run/build/vita3k/vita3k/packages/src/pkg.cpp:203:10: warning: ignoring return value of ‘size_t fread(void*, size_t, size_t, FILE*)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  203 |     fread(sfo_buffer.data(), sfo_buffer.size(), 1, infile);
      |     ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/run/build/vita3k/vita3k/packages/src/pkg.cpp:252:14: warning: ignoring return value of ‘size_t fread(void*, size_t, size_t, FILE*)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  252 |         fread(&entry, sizeof(PkgEntry), 1, infile);
      |         ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/run/build/vita3k/vita3k/packages/src/pkg.cpp:265:14: warning: ignoring return value of ‘size_t fread(void*, size_t, size_t, FILE*)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  265 |         fread(name.data(), byte_swap(entry.name_size), 1, infile);
      |         ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/run/build/vita3k/vita3k/packages/src/pkg.cpp:288:22: warning: ignoring return value of ‘size_t fread(void*, size_t, size_t, FILE*)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  288 |                 fread(buffer.data(), size, 1, infile);
      |                 ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/run/build/vita3k/vita3k/packages/src/pkg.cpp: In function ‘std::string find_pkg_zrif(const boost::filesystem::path&, const boost::filesystem::path&)’:
/run/build/vita3k/vita3k/packages/src/pkg.cpp:362:10: warning: ignoring return value of ‘size_t fread(void*, size_t, size_t, FILE*)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  362 |     fread(&pkg_header, sizeof(PkgHeader), 1, infile);
      |     ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[847/1239] Building CXX object vita3k/packages/CMakeFiles/packages.dir/src/pup.cpp.o
/run/build/vita3k/vita3k/packages/src/pup.cpp: In function ‘void extract_pup_files(const boost::filesystem::path&, const boost::filesystem::path&)’:
/run/build/vita3k/vita3k/packages/src/pup.cpp:117:10: warning: ignoring return value of ‘size_t fread(void*, size_t, size_t, FILE*)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  117 |     fread(header, SCEUF_HEADER_SIZE, 1, infile);
      |     ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/run/build/vita3k/vita3k/packages/src/pup.cpp:142:14: warning: ignoring return value of ‘size_t fread(void*, size_t, size_t, FILE*)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  142 |         fread(rec, SCEUF_FILEREC_SIZE, 1, infile);
      |         ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/run/build/vita3k/vita3k/packages/src/pup.cpp:160:18: warning: ignoring return value of ‘size_t fread(void*, size_t, size_t, FILE*)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  160 |             fread(hdr, HEADER_LENGTH, 1, infile);
      |             ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/run/build/vita3k/vita3k/packages/src/pup.cpp:167:14: warning: ignoring return value of ‘size_t fread(void*, size_t, size_t, FILE*)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  167 |         fread(buffer.data(), length, 1, infile);
      |         ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[993/1239] Building CXX object vita3k/modules/CMakeFiles/modules.dir/SceLibDbg/SceDbg.cpp.o
In file included from /run/build/vita3k/external/printf/v3kprintf.h:36,
                 from /run/build/vita3k/vita3k/modules/SceLibDbg/SceDbg.cpp:22:
/usr/include/c++/15.2.0/cstdbool:50:6: warning: #warning "<cstdbool> is deprecated in C++17, remove the #include" [-Wcpp]
   50 | #    warning "<cstdbool> is deprecated in C++17, remove the #include"
      |      ^~~~~~~
[1007/1239] Building CXX object vita3k/modules/CMakeFiles/modules.dir/SceLibc/SceLibc.cpp.o
In file included from /run/build/vita3k/external/printf/v3kprintf.h:36,
                 from /run/build/vita3k/vita3k/modules/SceLibc/SceLibc.cpp:27:
/usr/include/c++/15.2.0/cstdbool:50:6: warning: #warning "<cstdbool> is deprecated in C++17, remove the #include" [-Wcpp]
   50 | #    warning "<cstdbool> is deprecated in C++17, remove the #include"
      |      ^~~~~~~
[1040/1239] Building CXX object vita3k/modules/CMakeFiles/modules.dir/SceLibKernel/SceLibKernel.cpp.o
In file included from /run/build/vita3k/external/printf/v3kprintf.h:36,
                 from /run/build/vita3k/vita3k/modules/SceLibKernel/SceLibKernel.cpp:20:
/usr/include/c++/15.2.0/cstdbool:50:6: warning: #warning "<cstdbool> is deprecated in C++17, remove the #include" [-Wcpp]
   50 | #    warning "<cstdbool> is deprecated in C++17, remove the #include"
      |      ^~~~~~~
[1067/1239] Building CXX object vita3k/modules/CMakeFiles/modules.dir/SceProcessmgr/SceProcessmgr.cpp.o
/run/build/vita3k/vita3k/modules/SceProcessmgr/SceProcessmgr.cpp:298:51: warning: multi-character character constant [-Wmultichar]
  298 |     if (process_param && (process_param->magic == '2PSP') && (process_param->version != 0)) {
      |                                                   ^~~~~~
[1112/1239] Building CXX object vita3k/modules/CMakeFiles/modules.dir/SceSysmem/SceSysclibForDriver.cpp.o
In file included from /run/build/vita3k/external/printf/v3kprintf.h:36,
                 from /run/build/vita3k/vita3k/modules/SceSysmem/SceSysclibForDriver.cpp:24:
/usr/include/c++/15.2.0/cstdbool:50:6: warning: #warning "<cstdbool> is deprecated in C++17, remove the #include" [-Wcpp]
   50 | #    warning "<cstdbool> is deprecated in C++17, remove the #include"
      |      ^~~~~~~
[1145/1239] Building CXX object vita3k/emuenv/CMakeFiles/emuenv.dir/src/emuenv.cpp.o
In file included from /run/build/vita3k/vita3k/emuenv/src/emuenv.cpp:48:
/run/build/vita3k/vita3k/util/include/util/warning.h:24:61: warning: unknown option after ‘#pragma GCC diagnostic’ kind [-Wpragmas]
   24 | #define DISABLE_WARNING_NO_PUSH(warningNumber, warningName) _Pragma(STRINGIZE__(GCC diagnostic ignored warningName))
      |                                                             ^~~~~~~
/run/build/vita3k/vita3k/util/include/util/warning.h:36:5: note: in expansion of macro ‘DISABLE_WARNING_NO_PUSH’
   36 |     DISABLE_WARNING_NO_PUSH(warningNumber, warningName)
      |     ^~~~~~~~~~~~~~~~~~~~~~~
/run/build/vita3k/vita3k/emuenv/src/emuenv.cpp:52:1: note: in expansion of macro ‘DISABLE_WARNING_BEGIN’
   52 | DISABLE_WARNING_BEGIN(5038, "-Wreorder-ctor")
      | ^~~~~~~~~~~~~~~~~~~~~
/run/build/vita3k/vita3k/util/include/util/warning.h:24:61: note: did you mean ‘-Wreorder’?
   24 | #define DISABLE_WARNING_NO_PUSH(warningNumber, warningName) _Pragma(STRINGIZE__(GCC diagnostic ignored warningName))
      |                                                             ^~~~~~~
/run/build/vita3k/vita3k/util/include/util/warning.h:36:5: note: in expansion of macro ‘DISABLE_WARNING_NO_PUSH’
   36 |     DISABLE_WARNING_NO_PUSH(warningNumber, warningName)
      |     ^~~~~~~~~~~~~~~~~~~~~~~
/run/build/vita3k/vita3k/emuenv/src/emuenv.cpp:52:1: note: in expansion of macro ‘DISABLE_WARNING_BEGIN’
   52 | DISABLE_WARNING_BEGIN(5038, "-Wreorder-ctor")
      | ^~~~~~~~~~~~~~~~~~~~~
[1152/1239] Generating ../qt_translations/vita3k_en.qm
Updating '/run/build/vita3k/_flatpak_build/qt_translations/vita3k_en.qm'...
    Generated 0 translation(s) (0 finished and 0 unfinished)
    Ignored 687 untranslated source text(s)
[1199/1239] Building CXX object vita3k/gui-qt/CMakeFiles/gui-qt.dir/gui-qt_autogen/mocs_compilation.cpp.o
In file included from /run/build/vita3k/_flatpak_build/vita3k/gui-qt/gui-qt_autogen/6CKPAK4PGY/moc_trophy_collection_dialog.cpp:9,
                 from /run/build/vita3k/_flatpak_build/vita3k/gui-qt/gui-qt_autogen/mocs_compilation.cpp:26:
/run/build/vita3k/_flatpak_build/vita3k/gui-qt/gui-qt_autogen/6CKPAK4PGY/../../../../../vita3k/gui-qt/include/gui-qt/trophy_collection_dialog.h: In member function ‘void TrophyFilterProxy::invalidate()’:
/run/build/vita3k/_flatpak_build/vita3k/gui-qt/gui-qt_autogen/6CKPAK4PGY/../../../../../vita3k/gui-qt/include/gui-qt/trophy_collection_dialog.h:59:41: warning: ‘void QSortFilterProxyModel::invalidateFilter()’ is deprecated: Use begin/endFilterChange() instead [-Wdeprecated-declarations]
   59 |     void invalidate() { invalidateFilter(); }
      |                         ~~~~~~~~~~~~~~~~^~
In file included from /usr/include/QtCore/QSortFilterProxyModel:1,
                 from /run/build/vita3k/_flatpak_build/vita3k/gui-qt/gui-qt_autogen/6CKPAK4PGY/../../../../../vita3k/gui-qt/include/gui-qt/trophy_collection_dialog.h:25:
/usr/include/QtCore/qsortfilterproxymodel.h:128:10: note: declared here
  128 |     void invalidateFilter();
      |          ^~~~~~~~~~~~~~~~
[1210/1239] Building CXX object vita3k/gui-qt/CMakeFiles/gui-qt.dir/src/main_window.cpp.o
In file included from /run/build/vita3k/vita3k/gui-qt/src/main_window.cpp:44:
/run/build/vita3k/vita3k/gui-qt/include/gui-qt/trophy_collection_dialog.h: In member function ‘void TrophyFilterProxy::invalidate()’:
/run/build/vita3k/vita3k/gui-qt/include/gui-qt/trophy_collection_dialog.h:59:41: warning: ‘void QSortFilterProxyModel::invalidateFilter()’ is deprecated: Use begin/endFilterChange() instead [-Wdeprecated-declarations]
   59 |     void invalidate() { invalidateFilter(); }
      |                         ~~~~~~~~~~~~~~~~^~
In file included from /usr/include/QtCore/QSortFilterProxyModel:1,
                 from /run/build/vita3k/vita3k/gui-qt/include/gui-qt/trophy_collection_dialog.h:25:
/usr/include/QtCore/qsortfilterproxymodel.h:128:10: note: declared here
  128 |     void invalidateFilter();
      |          ^~~~~~~~~~~~~~~~
[1232/1239] Building CXX object vita3k/gui-qt/CMakeFiles/gui-qt.dir/src/trophy_collection_dialog.cpp.o
In file included from /run/build/vita3k/vita3k/gui-qt/src/trophy_collection_dialog.cpp:20:
/run/build/vita3k/vita3k/gui-qt/include/gui-qt/trophy_collection_dialog.h: In member function ‘void TrophyFilterProxy::invalidate()’:
/run/build/vita3k/vita3k/gui-qt/include/gui-qt/trophy_collection_dialog.h:59:41: warning: ‘void QSortFilterProxyModel::invalidateFilter()’ is deprecated: Use begin/endFilterChange() instead [-Wdeprecated-declarations]
   59 |     void invalidate() { invalidateFilter(); }
      |                         ~~~~~~~~~~~~~~~~^~
In file included from /usr/include/QtCore/QSortFilterProxyModel:1,
                 from /run/build/vita3k/vita3k/gui-qt/include/gui-qt/trophy_collection_dialog.h:25:
/usr/include/QtCore/qsortfilterproxymodel.h:128:10: note: declared here
  128 |     void invalidateFilter();
      |          ^~~~~~~~~~~~~~~~
[1238/1239] Linking CXX executable bin/Vita3K
FAILED: [code=1] bin/Vita3K 
: && /usr/bin/c++ -O2 -pipe -g -Wp,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -O2 -g -DNDEBUG -L/app/lib -Wl,-z,relro,-z,now -Wl,--as-needed -Wl,-rpath='$ORIGIN' -pthread -Wl,--dependency-file=vita3k/CMakeFiles/vita3k.dir/link.d vita3k/CMakeFiles/vita3k.dir/main.cpp.o vita3k/CMakeFiles/vita3k.dir/interface.cpp.o vita3k/CMakeFiles/vita3k.dir/performance.cpp.o -o bin/Vita3K  vita3k/app/libapp.a  vita3k/audio/libaudio.a  vita3k/compat/libcompat.a  vita3k/config/libconfig.a  external/libcppcommon.a  vita3k/ctrl/libctrl.a  vita3k/display/libdisplay.a  vita3k/ime/libime.a  vita3k/lang/liblang.a  vita3k/gdbstub/libgdbstub.a  vita3k/gxm/libgxm.a  vita3k/io/libio.a  external/libminiz.a  vita3k/modules/libmodules.a  vita3k/motion/libmotion.a  vita3k/packages/libpackages.a  vita3k/patch/libpatch.a  vita3k/renderer/librenderer.a  vita3k/shader/libshader.a  vita3k/touch/libtouch.a  vita3k/util/libutil.a  external/psvpfstools/psvpfsparser/cmake/libpsvpfsparser.a  vita3k/gui-qt/libgui-qt.a  external/libtracy.a  external/libcppcommon.a  vita3k/app/libapp.a  vita3k/compat/libcompat.a  vita3k/ctrl/libctrl.a  vita3k/display/libdisplay.a  vita3k/gdbstub/libgdbstub.a  vita3k/modules/libmodules.a  vita3k/motion/libmotion.a  vita3k/packages/libpackages.a  vita3k/renderer/librenderer.a  vita3k/touch/libtouch.a  vita3k/emuenv/libemuenv.a  vita3k/net/libnet.a  vita3k/module/libmodule.a  vita3k/overlay/liboverlay.a  vita3k/app/libapp.a  vita3k/compat/libcompat.a  vita3k/ctrl/libctrl.a  vita3k/display/libdisplay.a  vita3k/gdbstub/libgdbstub.a  vita3k/modules/libmodules.a  vita3k/motion/libmotion.a  vita3k/packages/libpackages.a  vita3k/renderer/librenderer.a  vita3k/touch/libtouch.a  vita3k/emuenv/libemuenv.a  vita3k/net/libnet.a  vita3k/module/libmodule.a  vita3k/overlay/liboverlay.a  vita3k/patch/libpatch.a  external/libdlmalloc.a  external/substitute/libsubstitute.a  external/psvpfstools/psvpfsparser/cmake/libpsvpfsparser.a  external/psvpfstools/libb64/liblibb64.a  /usr/lib/x86_64-linux-gnu/libz.so  external/libfat16/libFAT16.a  vita3k/shader/libshader.a  vita3k/gxm/libgxm.a  external/glslang/SPIRV/libSPIRV.a  external/glslang/glslang/libglslang.a  external/SPIRV-Cross/libspirv-cross-glsl.a  external/SPIRV-Cross/libspirv-cross-core.a  /usr/lib/x86_64-linux-gnu/libSM.so  /usr/lib/x86_64-linux-gnu/libICE.so  /usr/lib/x86_64-linux-gnu/libX11.so  /usr/lib/x86_64-linux-gnu/libXext.so  external/xxHash/cmake_unofficial/libxxhash.a  vita3k/glutil/libglutil.a  external/libglad.a  vita3k/vkutil/libvkutil.a  -lwayland-client  vita3k/audio/libaudio.a  external/cubeb/libcubeb.a  vita3k/camera/libcamera.a  vita3k/ngs/libngs.a  vita3k/codec/libcodec.a  external/liblibatrac9.a  /usr/lib/x86_64-linux-gnu/libavcodec.so  /usr/lib/x86_64-linux-gnu/libavformat.so  /usr/lib/x86_64-linux-gnu/libavutil.so  /usr/lib/x86_64-linux-gnu/libswscale.so  /usr/lib/x86_64-linux-gnu/libswresample.so  vita3k/regmgr/libregmgr.a  vita3k/ime/libime.a  vita3k/lang/liblang.a  vita3k/np/libnp.a  vita3k/io/libio.a  vita3k/kernel/libkernel.a  external/libminiz.a  vita3k/cpu/libcpu.a  external/dynarmic/src/dynarmic/libdynarmic.a  external/dynarmic/externals/zydis/libZydis.a  external/dynarmic/externals/zydis/zycore/libZycore.a  -lrt  external/dynarmic/externals/mcl/src/libmcl.a  vita3k/rtc/librtc.a  vita3k/nids/libnids.a  external/psvpfstools/libzrif/liblibzRIF.a  /usr/lib/x86_64-linux-gnu/libQt6Widgets.so.6.10.3  /usr/lib/x86_64-linux-gnu/libQt6Svg.so.6.10.3  /usr/lib/x86_64-linux-gnu/libQt6Multimedia.so.6.10.3  /usr/lib/x86_64-linux-gnu/libQt6Network.so.6.10.3  /usr/lib/x86_64-linux-gnu/libQt6Concurrent.so.6.10.3  /usr/lib/x86_64-linux-gnu/libQt6Gui.so.6.10.3  /usr/lib/x86_64-linux-gnu/libGLX.so  /usr/lib/x86_64-linux-gnu/libOpenGL.so  /usr/lib/x86_64-linux-gnu/libQt6Core.so.6.10.3  /usr/lib/x86_64-linux-gnu/libxkbcommon.so  vita3k/updater/libupdater.a  vita3k/config/libconfig.a  vita3k/util/libutil.a  vita3k/mem/libmem.a  vita3k/http/libhttp.a  vita3k/util/libutil.a  vita3k/mem/libmem.a  vita3k/http/libhttp.a  external/sdl/libSDL3.a  -lm  external/boost/lib/libboost_filesystem.a  external/boost/lib/libboost_atomic.a  external/spdlog/libspdlog.a  /usr/lib/x86_64-linux-gnu/libcurl.so.4.8.0  /usr/lib/x86_64-linux-gnu/libssl.so  /usr/lib/x86_64-linux-gnu/libcrypto.so  external/libtracy.a  -ldl  -lpthread  external/pugixml/libpugixml.a  vita3k/input/libinput.a  external/yaml-cpp/libyaml-cpp.a  external/fmt/libfmt.a && cd /run/build/vita3k/_flatpak_build/vita3k && /usr/bin/cmake -E copy_directory /run/build/vita3k/vita3k/../data /run/build/vita3k/_flatpak_build/bin/data && /usr/bin/cmake -E copy_directory /run/build/vita3k/_flatpak_build/qt_translations /run/build/vita3k/_flatpak_build/bin/translations && /usr/bin/cmake -E copy_directory /run/build/vita3k/vita3k/shaders-builtin /run/build/vita3k/_flatpak_build/bin/shaders-builtin && /usr/bin/cmake -E copy_directory /run/build/vita3k/vita3k/icons /run/build/vita3k/_flatpak_build/bin/icons && /usr/bin/cmake -E copy_if_different /run/build/vita3k/COPYING.txt /run/build/vita3k/_flatpak_build/bin/COPYING.txt && /usr/bin/cmake -E copy_if_different /run/build/vita3k/README.md /run/build/vita3k/_flatpak_build/bin/README.md
/usr/x86_64-unknown-linux-gnu/bin/ld.bfd: /usr/lib/x86_64-linux-gnu/libavcodec.so: undefined reference to symbol 'inflateEnd'
/usr/x86_64-unknown-linux-gnu/bin/ld.bfd: /usr/lib/x86_64-linux-gnu/libz.so: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
Error: module vita3k: Child process exited with code 1
</details>

**Comment by fpiesche** [comment]
> I think for verifying i would put it aside until we are 100% sure it can be maintained by someone trustable or is owned by the org, i dont think it would be a problem if its left unverified for some time

Notably some distributions (e.g. Mint) now hide non-verified Flatpaks from their graphical package managers. They can still be installed using the terminal and show up normally on the Flathub website, but Mint's "app store" style package manager will not show them in search results etc.

> Also considering vita3k doesnt have tagged release builds and builds happen pretty much all the time, how does updating work in it? Will flathub check for updates on the git and build automatically or how would it work?

There are automatic update checks that can be configured for Flatpak manifests and are run every 15 minutes on Flathub's repos. However, Flathub will not accept projects that don't have stable release builds, so a Flathub submission is probably dead in the water there :(

**Comment by j8r** [comment]
> Flathub will not accept projects that don't have stable release builds, so a Flathub submission is probably dead in the water there :(

That's in theory but it could be possible, we were able to keep RPCS3 rolling update through a adding a custom GitHub Action updater.
This avoids to strain the Flathub builder infrastructure and spam the Flathub repository with Pull Requests.

**Comment by fpiesche** [comment]
> [@fpiesche](https://github.com/fpiesche) Thanks for your work, unfortunately Vita3K changed since then and no longer compiles after having switched to Qt. Vita3K only support Qt 6.11, but for now only 6.10 is available in `org.kde.Sdk`. I tried to compile with 6.10 but got an error.

There is a 6.11 KDE SDK but it's currently on the beta branch only, and there doesn't seem to be a version of the LLVM extension that works with it. However I managed to get vita3k to build just fine against 6.10 by simply editing the minimum version specified in `cmake/qt6.cmake` - will need to do some testing to ensure this doesn't cause problems though.

The other option would be to wait for the 6.11 SDK to get promoted from the beta branch to stable.

**Comment by imWatchful** [comment]
Misclick

**Comment by EXtremeExploit** [comment]
If vita3k compiles just fine against 6.10 then it might be worth looking into what the actual minimum is so it compiles by default on most platforms in the present
But will probably have to be done in another PR

**Comment by fpiesche** [comment]
<img width="1300" height="767" alt="Image" src="https://github.com/user-attachments/assets/f6a6359f-2240-4302-b65c-b0b6cc2b0d4e" />

seems to do just fine on 6.10! any particular bits in the UI that might need testing? Had a poke around the things that are obviously accessible from the menus and settings dialog and everything looks in order there.

**Comment by imWatchful** [comment]
Keeping the Qt version the same across all builds will be in the best interests of everyone's sanity, so I'm not sure about lowering the minimum requirement...

**Comment by EXtremeExploit** [comment]
fpiesche probably means lowering the requirements globally, not for linux only

**Comment by imWatchful** [comment]
🗿 Me no like

**Comment by fpiesche** [comment]
In which case a Flatpak build would likely need to wait for the 6.11 SDK to leave beta - I doubt that Flathub would accept apps using beta runtimes on their stable branch.

For what it's worth, Qt 6.8 is the current LTS release with support until 2029; 6.11 will be EOL in March 2027.

**Comment by ttang4299** [comment]
Waiting for 6.11 to get promoted to stable seems like the pathway with the least amount of pushback. Has KDE shared an ETA for that? And what would be the next LTS release for Qt anyways?

**Comment by imWatchful** [comment]
I believe 6.12 will be the next LTS release so we'll move to that when appropriate

**Comment by fpiesche** [comment]
> Waiting for 6.11 to get promoted to stable seems like the pathway with the least amount of pushback. Has KDE shared an ETA for that? And what would be the next LTS release for Qt anyways?

I've checked in with the Flatpak Matrix chat to see if any of the maintainers over there know of a rough timeline. Will update if/when I hear back :)

**Comment by fpiesche** [comment]
Qt 6.11 Flatpak runtime is now public as `org.kde.Platform//6.11`!

**Comment by EXtremeExploit** [comment]
#4032 merged, so the only todo right now is the ffmpeg stuff and getting the build file actually setup and working with latest?
Im not sure about the ffmpeg one since according to macdu it might be better to not do it at all, it might not even be necessary at all who knows
